### PR TITLE
Added notifications documentation.

### PR DESCRIPTION
**Pull request by Hyperkid123**
Docs for notifications.

Solves https://github.com/RedHatInsights/insights-frontend-components/issues/142

### CODE REVIEW

**Comment by Hyperkid123** [comment]
cc @karelhala

File: doc/components/notifications.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+```javascript
+type: 'FOO_PENDING',
+type: 'FOO_FULFILLED',
+type: 'FOO_REJECTED'
+```
+
+Notifications middleware is checking actions for these suffixes and if defined, it will create a notification.
+
+### Configuration.
+Like in teh basic example, you must add your, reducer and notifications container. On to of that, you must add the middleware to your store:
```
Comment (karelhala) [inline doc/components/notifications.md]: lol `teh` :smile:

File: doc/components/notifications.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+```JSX
+import promiseMiddleware from 'redux-promise-middleware'
+import { ReducerRegistry } from '@red-hat-insights/insights-frontend-components/'
+import { ..., notificationsMiddleware } from '@red-hat-insights/insights-frontend-components/components/Notifications'
+
+const registry = new ReducerRegistry({}, [promiseMiddleware(), notificationsMiddleware()]);
```
Comment (karelhala) [inline doc/components/notifications.md]: Same as before, please use `getRegistry` function.

File: doc/components/notifications.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  errorDescriptionKey: 'detail' // path to notification description in error response
+}
+
+const registry = new ReducerRegistry({}, [..., notificationsMiddleware({...defaultOptions})]);
+
+```
+
+**Some useful options**
+
+1. Turning of automatic error notifications
```
Comment (karelhala) [inline doc/components/notifications.md]: `Turning of` -> `Turning off`

File: doc/components/notifications.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+You can use notifications without the middleware. For that you will need the `NotificationsPortal` component and add `notifications` reducer to your store (in the example below is used the insings redux registry).
+
+```JSX
+import React, { Fragment } from 'react'
+import { Provider } from 'react-redux'
+import { ReducerRegistry } from '@red-hat-insights/insights-frontend-components/'
+import { NotificationsPortal, notifications } from '@red-hat-insights/insights-frontend-components/components/Notifications'
+
+// you also need styles for the component
+import '@red-hat-insights/insights-frontend-components/components/Notifications.css'
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply doc/components/notifications.md]: I don't think that the scss files are published in build version. Or at least i did not find it there.

**Pull request merged**